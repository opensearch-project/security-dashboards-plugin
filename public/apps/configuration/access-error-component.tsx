/*
 *   Copyright OpenSearch Contributors
 *
 *   Licensed under the Apache License, Version 2.0 (the "License").
 *   You may not use this file except in compliance with the License.
 *   A copy of the License is located at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 *   or in the "license" file accompanying this file. This file is distributed
 *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 *   express or implied. See the License for the specific language governing
 *   permissions and limitations under the License.
 */

import { EuiLoadingContent, EuiPageContent } from '@elastic/eui';
import React from 'react';

interface AccessErrorComponentProps {
  loading?: boolean;
  dataSourceLabel?: string;
  message?: string;
}

export const AccessErrorComponent: React.FC<AccessErrorComponentProps> = (props) => {
  const {
    loading = false,
    dataSourceLabel,
    message = 'You do not have permissions to view this data',
  } = props;

  const displayMessage = message + (dataSourceLabel ? ` for ${props.dataSourceLabel}.` : '.');

  return loading ? <EuiLoadingContent /> : <EuiPageContent>{displayMessage}</EuiPageContent>;
};
