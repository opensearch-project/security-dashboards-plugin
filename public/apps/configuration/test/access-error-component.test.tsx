/*
 *   Copyright OpenSearch Contributors
 *
 *   Licensed under the Apache License, Version 2.0 (the "License").
 *   You may not use this file except in compliance with the License.
 *   A copy of the License is located at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 *   or in the "license" file accompanying this file. This file is distributed
 *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 *   express or implied. See the License for the specific language governing
 *   permissions and limitations under the License.
 */

import { shallow } from 'enzyme';
import React from 'react';
import { AccessErrorComponent } from '../access-error-component';

describe('AccessErrorComponent', () => {
  it('should render loading when loading the content', () => {
    const props = {
      dataSourceLabel: 'Test-cluster',
      loading: true,
    };
    const component = shallow(<AccessErrorComponent {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('should display default message prefix once loading is complete', () => {
    const props = {
      loading: false,
      dataSourceLabel: 'Test-cluster',
    };
    const component = shallow(<AccessErrorComponent {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('should display custom message prefix once loading is complete', () => {
    const props = {
      loading: false,
      dataSourceLabel: 'Test-cluster',
      message: 'Custom message prefix',
    };
    const component = shallow(<AccessErrorComponent {...props} />);
    expect(component).toMatchSnapshot();
  });
});
