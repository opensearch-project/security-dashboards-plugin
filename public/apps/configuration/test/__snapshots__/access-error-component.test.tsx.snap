// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccessErrorComponent should display custom message prefix once loading is complete 1`] = `
<EuiPageContent>
  Custom message prefix for Test-cluster.
</EuiPageContent>
`;

exports[`AccessErrorComponent should display default message prefix once loading is complete 1`] = `
<EuiPageContent>
  You do not have permissions to view this data for Test-cluster.
</EuiPageContent>
`;

exports[`AccessErrorComponent should render loading when loading the content 1`] = `<EuiLoadingContent />`;
